pkg/intermediate/aggregate.go

```diff
@@ -148,37 +169,97 @@ func (a *AggregationProcess) ForAllRecordsDo(callback FlowKeyRecordMapCallBack)
 	return nil
 }
 
-// GetLastUpdatedTimeOfFlow provides the last updated time in the format of IPFIX
-// field "flowEndSeconds".
-func (a *AggregationProcess) GetLastUpdatedTimeOfFlow(flowKey FlowKey) (uint32, error) {
+func (a *AggregationProcess) deleteFlowKeyFromMap(flowKey util.FlowKey) error {
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
-	record, exists := a.flowKeyRecordMap[flowKey]
+	return a.deleteFlowKeyFromMapWithoutLock(flowKey)
+}
+
+func (a *AggregationProcess) deleteFlowKeyFromMapWithoutLock(flowKey util.FlowKey) error {
+	_, exists := a.flowKeyRecordMap[flowKey]
 	if !exists {
-		return 0, fmt.Errorf("flow key is not present in the map")
+		return fmt.Errorf("flow key is not present in the map")
 	}
-	flowEndField, exists := record.Record.GetInfoElementWithValue("flowEndSeconds")
-	if exists {
-		return flowEndField.Value.(uint32), nil
-	} else {
-		return 0, fmt.Errorf("flowEndSeconds field is not present in the record")
+	delete(a.flowKeyRecordMap, flowKey)
+	return nil
+}
+
+// GetExpiryFromExpirePriorityQueue returns the earliest timestamp (active expiry
+// or inactive expiry) from expire priority queue.
+func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+
+	if a.expirePriorityQueue.Len() > 0 {
+		topItem := a.expirePriorityQueue.Peek()
+		if topItem.ActiveExpireTime.Before(topItem.InactiveExpireTime) {
+			return topItem.ActiveExpireTime.Sub(time.Now())
+		} else {
+			return topItem.InactiveExpireTime.Sub(time.Now())
```

Comment:
as a rule (applies to the code below as well) I suggest calling `time.Now()` a single time and storing the result in a variable

Reply:
Done.  Good suggestion.